backup-unit/browserunit.cpp

```diff
@@ -70,21 +72,67 @@ const QString browser_dir = ".local/share/org.sailfishos/browser";
 const QString moz_dir = browser_dir + "/.mozilla";
 // thumbs
 const QString cache_dir = ".cache/org.sailfishos/browser";
+// files
+const QString bookmarks = "/bookmarks.json";
+const QString database = "/sailfish-browser.sqlite";
+const QString keys = "/key3.db";
+const QString signons = "/signons.sqlite";
 
 const QVariantMap info = {
     {"home", QVariantMap({
                 {"data", QVariantList({
-                                  browser_dir + "/bookmarks.json"
+                                  browser_dir + bookmarks
                                 })}
                 , {"bin", QVariantList({
-                                  moz_dir + "/key3.db"
-                                , browser_dir + "/sailfish-browser.sqlite"
+                                  moz_dir + keys
+                                , browser_dir + database
                                 , cache_dir
-                                , moz_dir + "/signons.sqlite"
+                                , moz_dir + signons
                                 })
                         }})}
     , {"options", QVariantMap({{"overwrite", true}})}
 };
+
+// old paths
+const QString old_browser_dir = ".local/share/org.sailfishos/sailfish-browser";
+const QString old_moz_dir = ".mozilla/mozembed";
+const QString old_cache_dir = ".cache/org.sailfishos/sailfish-browser";
+
+}
+
+void fix(QString common, QString file, QString oldPath, QString newPath)
+{
+    const auto pathTemplate = QStringLiteral("%1/%3/%2").arg(common).arg(file);
+    QFileInfo source(pathTemplate.arg(oldPath));
+    if (source.exists()) {
+        QFileInfo dest(pathTemplate.arg(newPath));
+        dest.dir().mkpath(".");
+        if (!source.dir().rename(source.fileName(), dest.absoluteFilePath())) {
```

Comment:
I wonder if I should instead copy the files with vault::unit::copy so that it can keep the timestamps and such. I don't know if it works properly inside the archive though.

Reply:
Ah, it seems it might just keep those already.